api/services/inventory/sock_handler.go

```diff
@@ -175,6 +177,72 @@ func (h *SockHandler) handleGetSockDetails(w http.ResponseWriter, r *http.Reques
 	utils.WriteJson(w, http.StatusOK, sock)
 }
 
+// @Summary Updates a sock and its variants
+// @Description Updates an existing sock's details and its variants.
+// @Tags Socks
+// @Accept json
+// @Produce json
+// @Security Bearer
+// @Param sock_id path int true "Sock ID"
+// @Param sock body types.UpdateSockRequest true "Sock update details"
+// @Success 200 {object} types.Message
+// @Router /socks/{sock_id} [patch]
+func (h *SockHandler) handleUpdateSock(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	sockIDstr := vars["sock_id"]
+
+	sockID, err := strconv.Atoi(sockIDstr)
+	if err != nil {
+		utils.WriteError(w, http.StatusBadRequest, errors.New("invalid sock ID"))
+		return
+	}
+
+	exists, err := h.store.SockExistsByID(sockID)
+	if err != nil {
+		utils.WriteError(w, http.StatusInternalServerError, err)
+		return
+	}
+
+	if !exists {
+		utils.WriteError(w, http.StatusNotFound, errors.New("sock not found"))
+		return
+	}
+
+	var req types.UpdateSockRequest
+	if err := utils.ParseJson(r, &req); err != nil {
+		utils.WriteError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	if err := utils.Validate.Struct(req); err != nil {
+		utils.WriteError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	// Convert the UpdateSockRequest to the types.Sock entity
+	sock := types.Sock{
```

Comment:
1. Make the changes I suggested in the `entities.go` file (and updates for `UpdateSockRequest`)
2. With the new definition of `UpdateSockRequest`, the code should more or less look like this:

```go
    sock := toSock(req.Sock)
	variants := toSockVariantArray(req.Variants)

	if err := h.store.UpdateSock(sockID, sock, variants); err != nil {
		utils.WriteError(w, http.StatusInternalServerError, err)
		return
	}
//...
```

This way you reuse the help functions :)

Reply:
Thank you! Fixed